### ISSUE TITLE
Support SAS tokens in Custom Script Extension

### ISSUE THREAD

**Issue by jjjordanmsft** [Azure/azure-linux-extensions#105]
It would be nice if the custom script extension accepted SAS tokens instead of forcing me to send the full storage account key (effectively giving the VM root access to the storage account); it would better fit the principle of least privilege.

The current issue is that my SAS key would be visible to anyone looking at a deployment log.  This means the extension should accept the fileUris parameter in protected settings.

**Comment by bingosummer** [comment]
Good idea. Thanks for your suggestions. @jjjordanmsft 
Will talk about it with our team, and give you feedbacks later.

**Comment by omerlh** [comment]
+1 on that feature

**Comment by boumenot** [comment]
cc: @ahmetalpbalkan - is this supported in the Microsoft.Azure.Extensions.CustomScript?

**Comment by ahmetalpbalkan** [comment]
@boumenot  No. This would change the design quite a bit (and we'd have to implement it on Windows as well).

I see that this feature would be valuable, however you can create a separate account for your scripts etc to obtain least privilege etc. This would be the current solution I can suggest. Also the storage credentials user provides in protectedSettings are always stored and transmitted with encryption.

For now, I am a bit conservative about adding a feature about this. Ideally in the future we will have some means of instance identity which would be ACL'd to access certain storage accounts and if the account of the specified storage blob is ACL'd for the VM, user could get the job done _without_ SAS or Account Keys.

That’s why for now I would love to steer clear from adding new features as in the future there will be cleaner way(s) to solve this problem.

**Comment by ejarvi** [comment]
Here is a scrappy solution I put together some time ago that combines SAS URL's with CSE: 

https://github.com/ejarvi/ade-cli-getting-started/blob/master/validate.sh#L182-L229

This bash script wires together Curl, CLI, SAS URL's, JSON config file params, and finally the custom script extension to accomplish what I think is a similar objective to what is being asked here.   First it creates a one time SAS URL pointing to the script that is to be run (input).   Next it creates a SAS URL that the script itself can use to write output to once it is running within the VM (output).  Then, it creates two local JSON files (public and protected settings for the custom script extension) with these SAS URL's embedded.  Then it calls the custom script extension. 

Just offering that example as one SAS oriented approach that's do-able today, and doesn't require any changes to the extension.  It does require some fancy footwork before calling the extension.

**Comment by miniGweek** [comment]
Current workaround which works for me - as mentioned here https://stackoverflow.com/questions/43752262/azure-vm-custom-script-extension-sas-token-support
Appending &sr=b at the end of the URL seems to get it working. strange!
